ledger/eval.go

```diff
@@ -1119,6 +1245,11 @@ func (eval *BlockEvaluator) GenerateBlock() (*ValidatedBlock, error) {
 		return nil, err
 	}
 
+	err = eval.modifyOfflineAccounts()
```

Comment:
could you move this call to `eval.finalValidation()`, just instead of "return nil" at the end ?
see https://github.com/algorand/go-algorand/pull/2922 for the rationale..